libcst/metadata/type_inference_provider.py

```diff
@@ -0,0 +1,38 @@
+from typing import Optional
+
+import libcst as cst
+from libcst import BatchableMetadataProvider
+from libcst.metadata import PositionProvider
+
+
+class TypeInferenceProvider(BatchableMetadataProvider[str]):
+    METADATA_DEPENDENCIES = (PositionProvider,)
+    is_cache_required = True
+
+    def __init__(self, cache: object):
+        super().__init__(cache)
+        lookup = {}
+        for item in cache:
+            location = item["location"]
+            start = location["start"]
+            end = location["stop"]
+            lookup[(start["line"], start["column"], end["line"], end["column"])] = item[
+                "annotation"
+            ]
+        self.lookup = lookup
+
+    def _parse_metadata(self, node: cst.CSTNode) -> None:
+        range = self.get_metadata(PositionProvider, node)
+        key = (range.start.line, range.start.column, range.end.line, range.end.column)
+        if key in self.lookup:
+            self.set_metadata(node, self.lookup[key])
+            self.lookup.pop(key)
```

Comment:
This is a weird use of pop. If you don't care about the return value, the more idiomatic pattern for dicts seems to be

```
del self.lookup[key]
```

(discarding the return value of pop on a list is common, but I think that's just because there's not as good of an alternative)

Reply:
I can just combine the two statements together.